src/_RegistryProvider.ts

```diff
@@ -0,0 +1,77 @@
+import Promise from 'dojo-core/Promise';
+
+import {
+	ActionLike,
+	CombinedRegistry,
+	StoreLike,
+	WidgetLike
+} from './createApp';
+
+/**
+ * Registry to (asynchronously) get instances by their ID.
+ */
+export interface Registry<T> {
```

Comment:
Hmm. While exported this is really an internal interface (e.g. I don't expect anyone to import it from `dojo-app/_RegistryProvider`. It's also more restricted than `IdentityRegistry` as the identifiers can only be strings, not symbols.